File: QE-Selenium-Java/Activity14.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity14 {
    public static void main(String[] args) {

        FirefoxDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/tables");
        System.out.println(driver.getTitle());

        java.util.List<WebElement> rows = driver.findElements(By.xpath("//table/tbody/tr"));
        System.out.println(rows.size());

        java.util.List<WebElement> cols = driver.findElements(By.xpath("//table/thead/tr/th"));
        System.out.println(cols.size());

        WebElement book = driver.findElement(By.xpath("//table/tbody/tr[5]/td[2]"));
        System.out.println(book.getText());

        WebElement price_header = driver.findElement(By.xpath("//table/thead/tr/th[contains(text(),'Price')]"));
        price_header.click();

        WebElement book_af_sort = driver.findElement(By.xpath("//table/tbody/tr[5]/td[2]"));
        System.out.println(book_af_sort.getText());

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/tables") 
print(driver.title)
rows=driver.find_elements(By.XPATH,"//table/tbody/tr")
print(len(rows))
cols=driver.find_elements(By.XPATH,"//table/thead/tr/th")
print(len(cols))
book=driver.find_element(By.XPATH,"//table/tbody/tr[5]/td[2]")
print(book.text)
price_header=driver.find_element(By.XPATH,"//table/thead/tr/th[contains(text(),'Price')]").click()
book_af_sort=driver.find_element(By.XPATH,"//table/tbody/tr[5]/td[2]")
print(book_af_sort.text)
driver.quit()
